cmd/collector/collector.go

```diff
@@ -60,7 +66,9 @@ func addIPFIXFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&IPFIXTransport, "ipfix.transport", "tcp", "IPFIX collector transport layer")
 }
 
-func printIPFIXMessage(msg *entities.Message) {
+func addIPFIXMessage(msg *entities.Message) {
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
it's much better if you only lock the mutex when you need it, i.e. for the last statement in this function

Reply:
Updated.